File: data/imported/knowledge/slides/python/merged/snippets/snippet_021.py
# Adapted from aspose.org: knowledge/slides/python/merged/snippets/snippet_021.py @ 7f72da4e1423546104b40fa8cebf5b9ae3ce9c91
# Imported under the authorization recorded in
# plans/investigations/evidence/imported-corpus-v1/licensing-resolution-state.md

def test_behaviors_iterable(self):

        with Presentation() as pres:

            slide = pres.slides[0]

            shape = slide.shapes.add_auto_shape(ShapeType.RECTANGLE, 10, 10, 100, 50)

            effect = slide.timeline.main_sequence.add_effect(

                shape, anim.EffectType.FADE,

                anim.EffectSubtype.NONE,

                anim.EffectTriggerType.ON_CLICK,

            )

            items = list(effect.behaviors)

            assert len(items) == effect.behaviors.count

File: data/imported/knowledge/slides/cpp/merged/snippets/snippet_021.cpp
TEST_F(EffectFormatIntegrationTest, SoftEdge) {
    Presentation pres;
    auto& slide = clear_slide(pres);
    auto& shape = slide.shapes().add_auto_shape(ShapeType::RECTANGLE, 100, 100, 200, 100);
    auto& ef = shape.effect_format();
    ef.enable_soft_edge_effect();
    auto* se = ef.soft_edge_effect();
    ASSERT_NE(se, nullptr);
    se->set_radius(10);

    auto pres2 = save_and_reopen(pres);
    auto* se2 = pres2.slides()[0].shapes()[0].effect_format().soft_edge_effect();
    ASSERT_NE(se2, nullptr) << "soft_edge_effect should not be None after reload";
    EXPECT_EQ(se2->radius(), 10);
}